tests/test_md.py

```diff
@@ -391,6 +377,76 @@ def test_multiple_steps_local_validation(self):
         with pytest.raises(RuntimeError, match="k must be less than than 1000000.0"):
             ctxt.multiple_steps_local(100, np.array([1], dtype=np.int32), k=1e7)
 
+    def test_multiple_steps_local_selection_validation(self):
+        seed = 2022
+        np.random.seed(seed)
+
+        N = 8
+        D = 3
+
+        coords = np.random.rand(N, D).astype(dtype=np.float64) * 2
+        box = np.eye(3) * 3.0
+        masses = np.random.rand(N)
+
+        E = 2
+
+        params, potential = prepare_nb_system(
+            coords,
+            E,
+            p_scale=3.0,
+            cutoff=1.0,
+        )
+        nb_pot = potential.to_gpu(np.float32)
+
+        dt = 1.5e-3
+        radius = 1.2
+
+        v0 = np.zeros_like(coords)
+        bps = [nb_pot.bind(params).bound_impl]
+
+        # Compatible with local NVE since multiple_steps_local_selection doesn't depend on temperature
+        verlet = VelocityVerletIntegrator(dt, masses)
+        intg_impl = verlet.impl()
+
+        reference_idx = 0
+
+        # Verify that indices are correctly checked
+        ctxt = custom_ops.Context(coords, v0, box, intg_impl, bps)
+        with pytest.raises(RuntimeError, match="indices can't be empty"):
+            ctxt.multiple_steps_local_selection(100, reference_idx, np.array([], dtype=np.int32), radius=radius)
+
+        with pytest.raises(RuntimeError, match="index values must be less than N"):
+            ctxt.multiple_steps_local_selection(100, reference_idx, np.array([N * 2], dtype=np.int32), radius=radius)
+
+        with pytest.raises(RuntimeError, match="index values must be greater or equal to zero"):
+            ctxt.multiple_steps_local_selection(100, reference_idx, np.array([-1], dtype=np.int32), radius=radius)
+
+        with pytest.raises(RuntimeError, match="atom indices must be unique"):
+            ctxt.multiple_steps_local_selection(100, reference_idx, np.array([1, 1], dtype=np.int32), radius=radius)
+
+        with pytest.raises(RuntimeError, match="burn in steps must be greater or equal to zero"):
+            ctxt.multiple_steps_local_selection(
+                100, reference_idx, np.array([1], dtype=np.int32), radius=radius, burn_in=-5
+            )
+
+        with pytest.raises(RuntimeError, match="radius must be greater or equal to 0.1"):
+            ctxt.multiple_steps_local_selection(100, reference_idx, np.array([1], dtype=np.int32), radius=0.01)
+
+        with pytest.raises(RuntimeError, match="k must be at least one"):
+            ctxt.multiple_steps_local_selection(100, reference_idx, np.array([1], dtype=np.int32), k=0.0)
+
+        with pytest.raises(RuntimeError, match="k must be less than than 1000000.0"):
```

Comment:
nit: readability:
```suggestion
        with pytest.raises(RuntimeError, match="k must be less than than 1e6"):
```

Reply:
Since this is matching the value that comes out of C++ from https://github.com/proteneer/timemachine/blob/feature/local-selection-mask-md/timemachine/cpp/src/local_md_utils.cu#L160 would need to change that as well. Will look at converting the double to scientific notation.

Reply:
Ah, I see. Sorry, this comment was a bit too hasty. If it's easy to change, that would be great, but I don't think it's a big deal.

Reply:
Changed in [dbe3593](https://github.com/proteneer/timemachine/pull/1005/commits/dbe359392358e7c4893c838f9c40ff338b995655)